learning/tour-of-beam/backend/internal/storage/datastore.go

```diff
@@ -227,5 +227,34 @@ func (d *DatastoreDb) SaveContentTrees(ctx context.Context, trees []tob.ContentT
 	return nil
 }
 
+// Get learning unit content by unitId
+func (d *DatastoreDb) GetUnitContent(ctx context.Context, sdk tob.Sdk, unitId string) (unit *tob.Unit, err error) {
+	var tbNodes []TbLearningNode
+	rootKey := pgNameKey(TbLearningPathKind, sdkToKey(sdk), nil)
+
+	query := datastore.NewQuery(TbLearningNodeKind).
+		Namespace(PgNamespace).
+		Ancestor(rootKey).
+		FilterField("id", "=", unitId)
+
+	_, err = d.Client.GetAll(ctx, query, &tbNodes)
+	if err != nil {
+		return nil, fmt.Errorf("query unit failed: %w", err)
+	}
+
+	switch {
+	case len(tbNodes) == 0:
+		return nil, nil
```

Comment:
From my perspective, it's not good approach to return the double nil to handle this case later. I recommend you return error here.

As for the next step:
```
	if err != nil {
		log.Println("Get unit content error:", err)
		finalizeErrResponse(w, http.StatusInternalServerError, INTERNAL_ERROR, "storage error")
		return
	}
	if unit == nil {
		log.Println("Get unit content error:", err)
		finalizeErrResponse(w, http.StatusNotFound, NOT_FOUND, "unit not found")
		return
	}
```
Maybe the best way to create custom errors and handle them using switch-case

It's  a not major because it's related to your code style. I'm sharing to you my thoughts about that.

Reply:
I thought on it and decided to introduce an error in a service layer, where it belongs
(non-existing unit is OK in the database context)

Reply:
Ok, make sense, lgtm